src/legacy/core_plugins/expressions/public/expressions/lib/loader.ts

```diff
@@ -58,21 +58,22 @@ export class ExpressionLoader {
     });
 
     this.execute(expression, params);
-    // @ts-ignore
-    this.dataHandler = this.dataHandler;
   }
 
   destroy() {}
 
   cancel() {
+    if (!this.dataHandler) throw new Error('dataHandler not available.');
```

Comment:
dataHandler gets created inside execute function which is called in constructor, so its always available.

maybe extract dataHandler creation out of execute function directly in the constructor to avoid ts not figuring out its set

Reply:
Using not null assertion for this.